### DISCUSSION TITLE
Versions Compatible with Bitwarden Mobile Clients?

### DISCUSSION THREAD

**Discussion by f1d094** [dani-garcia/vaultwarden#4642] [General]
Sometime in the past couple months I was browsing the VW repo and saw a discussion about some changes coming to BW clients that would need to be considered because they would break things. I failed to finish reading and I forgot to bookmark the conversation. Now I cannot find it.

We are running 1.25.2 and I'm concerned that our mobile clients will unexpectedly stop working with our VaultWarden. I do plan to upgrade but just don't have the bandwidth right now (our deployment is a minor fork of 1.25.2 with some changes for our setup) Can anyone tell me if there are any breaking changes expected on the BW mobile clients anytime soon? More specifically, anything that would cause issues with v1.25.2?

If you know the Issue# that I originally stumbled across that would be great too...I'd like to follow.

**Comment by BlackDex** [comment]
You probably saw #4386, which only is for the new native mobile clients which are currently in beta stage.

**Comment by f1d094** [comment]
@BlackDex Will the new native Bitwarden clients currently in Beta still work with older versions of vaultwarden? Specifically v1.25.2? I'd like to know if I'm on a clock to upgrade before clients suddenly stop working. We have auto-updates enabled for apps on our mobile devices.

**Reply by BlackDex** [reply]
I'm actually surprised that all is still working with your version of almost 2 years old. Especially if all the clients are updated. 

But, once the new native clients are going to be mainstream, it will all stop working. Clients will not be able to login anymore, at least, not the Android and iOS clients. 

Also, what is holding you back from en update?

**Reply by f1d094** [reply]
I made a bunch of custom changes to the landing/login page that I've not scripted, properly documented, or made a proper diff of. Basically stripping the login page of any identifying information, changing the user/password errors to be more opaque ("Either the username or passphrase you entered is incorrect") and removing any new account options. IIRC I also obfuscated some of the variables and any references to the project in the visible login page source. The idea being that an attacker would have to work much harder to identify what they are trying to attack...and hopefully stopping cold any automated systems.

Hearing the above, I will make it a priority.

**Reply by BlackDex** [reply]
An attacker just goes to the `/api/config` page, has the version numbers (which the clients also need) and then just calls the api endpoints manually. Obfuscation and removal of those items doesn't make it not working at all anymore.

Also there are several settings to prevent account creation.

**Reply by f1d094** [reply]
TBH, I did not know about the api/config page when I first did the obfuscation...it is just a normal practice for anything I setup. Anyone specifically looking for vaultwarden would just look for the above, of course, but stripping the landing page helps stop the immediate identification from scraping bots just collecting future-target data or operating without prior knowledge. Coupled with fail2ban it helps reduce what winds up broadly visible on the Internet. FWIW, my VW instance does not show up for my IP on https://shodan.io

I presume the /api/config page is necessary for clients pre-auth?
